code/web/src/modules/user/api/actions.js

```diff
@@ -1,145 +1,175 @@
+import { routeApi } from "../../../setup/routes";
 
 // Actions Types
-export const LOGIN_REQUEST = 'AUTH/LOGIN_REQUEST'
-export const LOGIN_RESPONSE = 'AUTH/LOGIN_RESPONSE'
-export const SET_USER = 'AUTH/SET_USER'
-export const LOGOUT = 'AUTH/LOGOUT'
-export const CHANGE_EDIT_MODE = 'AUTH/CHANGE_EDIT_MODE'
-export const SAVE_PROFILE = 'AUTH/SAVE_PROFILE'
+export const LOGIN_REQUEST = "AUTH/LOGIN_REQUEST";
+export const LOGIN_RESPONSE = "AUTH/LOGIN_RESPONSE";
+export const SET_USER = "AUTH/SET_USER";
+export const LOGOUT = "AUTH/LOGOUT";
+export const CHANGE_EDIT_MODE = "AUTH/CHANGE_EDIT_MODE";
+export const SAVE_PROFILE = "AUTH/SAVE_PROFILE";
 
 // Actions
 
 //Set edit mode for user
 
 export function changeEditMode(user) {
-  return dispatch => {
+  return (dispatch) => {
     dispatch({
       type: CHANGE_EDIT_MODE,
-      user
-    })
-  }
+      user,
+    });
+  };
 }
 
 // Set a user after login or using localStorage token
 export function setUser(token, user) {
   if (token) {
-    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+    axios.defaults.headers.common["Authorization"] = `Bearer ${token}`;
   } else {
-    delete axios.defaults.headers.common['Authorization'];
+    delete axios.defaults.headers.common["Authorization"];
   }
 
-  return { type: SET_USER, user }
+  return { type: SET_USER, user };
 }
 
-export function saveProfile(updatedDetails) {
-  return dispatch => {
+export function saveProfile(updatedDetails, subDetails, userSubs) {
+  if (subDetails.nextDeliveryDate) {
+    userSubs.forEach((sub) => {
+      subDetails.id = sub
+      axios.post(
+        routeApi,
+        mutation({
+          operation: "subscriptionUpdate",
+          variables: subDetails,
+          fields: ["id"],
+        })
+      );
+    });
+  }
+  return (dispatch) => {
+    axios.post(
+      routeApi,
+      mutation({
+        operation: "userUpdate",
+        variables: updatedDetails,
+        fields: ["id"],
+      })
+    );
     dispatch({
       type: SAVE_PROFILE,
       email: updatedDetails.email,
-      img: updatedDetails.img,
+      img: updatedDetails.image,
       description: updatedDetails.description,
-      address: updatedDetails.address
-    })
-  }
+      address: updatedDetails.address,
+    });
+  };
 }
 
 // Login a user using credentials
 export function login(userCredentials, isLoading = true) {
-  return dispatch => {
+  return (dispatch) => {
     dispatch({
       type: LOGIN_REQUEST,
-      isLoading
-    })
-
-    return axios.post(routeApi, query({
-      operation: 'userLogin',
-      variables: userCredentials,
-      fields: ['user {name, email, role}', 'token']
-    }))
-      .then(response => {
-        let error = ''
+      isLoading,
+    });
+
+    return axios
+      .post(
+        routeApi,
+        query({
+          operation: "userLogin",
+          variables: userCredentials,
+          fields: ["user {name, email, role, id}", "token"],
```

Comment:
Also, I'm not seeing that the user has an `id` on my branch, is this something that backend has taken care of but hasn't been merged in yet?

Reply:
So make sure you logout then log back in, that should be enough to get the User's ID locally I believe